lib/send_messages/send_messages_request.js

```diff
@@ -25,10 +25,22 @@ module.exports = function(userId, password, content, recipients, callback) {
 
   recipients.forEach(function(recipient, index) {
     var _recipient = subElement(_recipients, 'ns:recipient');
-    _recipient.attrib.uid = index;
+    if (messageId) {
+      _recipient.attrib.uid = messageId;
+    } else {
+      _recipient.attrib.uid = index;
```

Comment:
Even if there are 5 recipients? i thought this will set it to the index of the loop